decode_map.go

```diff
@@ -226,7 +226,7 @@ func decodeStructValue(d *Decoder, strct reflect.Value) error {
 		return nil
 	}
 
-	fields := structs.Fields(strct.Type())
+	fields := structs.Fields(strct.Type(), false)
```

Comment:
This is the only place I don't like in this change, but I don't have a place to get the `msgpack` vs `json` tag during the decoding. Maybe introduce `Decoder.UseJSONTag` as well?